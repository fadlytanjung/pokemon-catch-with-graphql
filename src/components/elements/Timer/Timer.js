import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Timer(props) {
  const { className, time } = props;
  const customClass = [styles.root, className].filter(Boolean).join(' ');
  return (
    <span className={customClass}>
      <img alt="" src="/assets/ic_timer.svg" />{time[1]}:{time[2]}
    </span>
  );
}

Timer.defaultProps = {
  className: '',
  time: ['', '', ''],
};

Timer.propTypes = {
  className: PropTypes.string,
  time: PropTypes.arrayOf(PropTypes.string),
};
