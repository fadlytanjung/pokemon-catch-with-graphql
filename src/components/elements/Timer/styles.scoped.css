.root {
  align-items: center;
  background-color: #115b79;
  border-radius: 1.125rem;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  color: #ffffff;
  display: flex;
  padding: 0.25rem 0.5rem;
}

.root > img {
  height: 1rem;
  margin-right: 0.5rem;
}
