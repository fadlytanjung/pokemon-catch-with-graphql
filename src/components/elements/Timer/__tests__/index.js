import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Timer from '../Timer';

describe('src/components/elements/Timer', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Timer />);
    expect(tree).toMatchSnapshot();
  });
});
