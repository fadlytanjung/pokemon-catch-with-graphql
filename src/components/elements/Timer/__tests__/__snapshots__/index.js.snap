// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Timer render 1`] = `
<span
  className=""
>
  <img
    alt=""
    src="/assets/ic_timer.svg"
  />
  
  :
  
</span>
`;
